src/components/text-editor/prosemirror-adapter/plugins/image/inserter.ts

```
@@ -261,19 +283,253 @@
+};
 
-    return files.length > 0;
+/**
+ * Creates a smaller thumbnail from an image data URL
+ * @param dataUrl - Original image data URL
+ * @param maxWidth - Maximum width of thumbnail
+ * @param maxHeight - Maximum height of thumbnail
+ * @param quality - JPEG quality (0-1)
+ * @returns Promise resolving to a smaller thumbnail data URL
+ */
+const createOptimizedThumbnail = (
```

ESLint

[error] 'createOptimizedThumbnail' is assigned a value but never used. Allowed unused vars must match /^h$/u.

(@typescript-eslint/no-unused-vars)